ui/src/app/pages/admin/user/admin-user-profile.spec.tsx

```diff
@@ -114,7 +114,61 @@ describe('AdminUserProfile', () => {
     await waitUntilPageLoaded();
   });
 
-  it("should display the user's name, username, and initial credits usage", async () => {
+  it.each([true, false])(
+    "should display the user's name, username, initial credits information when enableInitialCreditsExpiration is %s",
+    async (enableInitialCreditsExpiration) => {
+      const givenName = 'John Q';
+      const familyName = 'Public';
+      const expectedFullName = 'John Q Public';
+
+      const username = 'some-email@yahoo.com';
+
+      const freeTierUsage = 543.21;
+      const freeTierDollarQuota = 678.99;
+      const expectedCreditsText = '$543.21 used of $678.99 limit';
+      const initialCreditsExpirationEpochMillis = Date.now();
+
+      serverConfigStore.set({
+        config: {
+          ...defaultServerConfig,
+          enableInitialCreditsExpiration,
+        },
+      });
+
+      updateTargetProfile({
+        username,
+        givenName,
+        familyName,
+        freeTierUsage,
+        freeTierDollarQuota,
+        initialCreditsExpirationEpochMillis,
+      });
+
+      component();
+      await waitUntilPageLoaded();
+      expect(
+        within(screen.getByTestId('name')).getByText(expectedFullName)
+      ).toBeInTheDocument();
+      expect(
+        within(screen.getByTestId('user-name')).getByText(username)
+      ).toBeInTheDocument();
+      expect(
+        within(screen.getByTestId('initial-credits-used')).getByText(
+          expectedCreditsText
+        )
+      ).toBeInTheDocument();
+
+      if (enableInitialCreditsExpiration) {
```

Comment:
I cannot think of a more efficient way of testing the presence or absence of this term in a parametrized way. I am open to suggestions.